### PR TITLE
creating our work section 

### LINKED ISSUE #20: Create "Our work" section
In this section all the past work done will be briefly described

### PR DESCRIPTION
**Pull request by rajitha1998**
Fixes #20 

## Proposed Changes

  - Creating project cards 
  - Setting the number of cards based on screen size
  - Switch card functionality 
  - Project pop up cards 
  - Mobile responsiveness 

## Screenshots

Video-_4-25-2020-10-57-48-PM_

### CODE REVIEW

**Comment by rajitha1998** [comment]
Build failing because of this [issue in is-promise](https://github.com/then/is-promise/issues/13)  😞@janethavi @hasalfernando @nandulaperera @Purushothh

**Comment by hasalfernando** [comment]
@rajitha1998 seems the issue isn't resolved yet. Let's wait for it to be resolved. However, the issue you mentioned isn't the active one. It is currently closed. [This](https://github.com/then/is-promise/issues/20) is the active issue thread.

**Review by nandulaperera — APPROVED** [review]
Nice work @rajitha1998

File: src/app/components/our-work/our-work.component.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+                                github: project.github|| null})">
+                            See more</button>
+                    </div>
+                </div>
+                <button type="button" class="slider_button btn btn-danger btn-just-icon" (click)="foward()"><i
+                        class="fa fa-angle-right"></i></button>
+            </div>
+        </div>
+    </div>
+</div>
```
Comment (janethavi) [inline src/app/components/our-work/our-work.component.html]: Shall we add an EOF line here?

File: src/app/components/project-card/project-card.component.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        </a>
+      </div>
+    </div>
+  </div>
+  <div class="modal-footer no-border-footer">
+    <div class="project-close mx-auto">
+      <button type="button" #closebutton class="btn btn btn-outline-primary btn-round" data-dismiss="modal" (click)="close()">Close</button>
+    </div>
+  </div>
+</div>
```
Comment (janethavi) [inline src/app/components/project-card/project-card.component.html]: Shall we add an EOF line here?

File: src/assets/json/projects.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    "github": "https://github.com/codeJunctionTeam/codeJunction-site"
+  },
+  {
+    "projectName": "Fire call",
+    "image": "firecall.jpg",
+    "description": "A sound wave classifying tool which analyse audio content. (Open Source ❤️)",
+    "largeDescription" : "Fire call is an open-source project which is developed using python and flask framework. This project aims to classify sound waves through analysing audio content. We welcome everyone to contribute in this project. To get started please visit our Github repository.",
+    "github": "https://github.com/codeJunctionTeam/fire-call"
+  }
+]
```
Comment (janethavi) [inline src/assets/json/projects.json]: Shall we add an EOF line here?

File: src/app/components/components.component.html [hunk 4]
```diff
@@ -15,7 +15,8 @@ <h2 class="presentation-subtitle text-center">Think | Search | Build</h2>
...
     </div>
     <div class="main">
-      <team-members></team-members>
-      <app-our-team></app-our-team>
+        <app-our-work></app-our-work>
+        <team-members></team-members>
+        <app-our-team></app-our-team>
     </div>
-</div>
+</div>
```
Comment (janethavi) [inline src/app/components/components.component.html]: Shall we add an EOF line here?

File: src/app/components/our-work/our-work.component.html [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<div class="section">
+    <div class="container">
+        <h2>Our Work</h2>
+        <div class="work-section">
+            <div class="row justify-content-center">
+                <button type="button" class="slider_button btn btn-danger btn-just-icon" (click)="backward()"><i
+                        class="	fa fa-angle-left"></i></button>
```
Comment (janethavi) [inline src/app/components/our-work/our-work.component.html]: Shall we remove the extra space in the CSS class value?

File: src/app/components/components.component.ts [hunk 6]
```diff
@@ -17,29 +17,30 @@ export class ComponentsComponent implements OnInit {
...
-            input_group[i].children[0].addEventListener('blur', function (){
+            input_group[i].children[0].addEventListener('blur', function () {
                 input_group[i].classList.remove('input-group-focus');
             });
         }
     }
-
 }
+
+
```
Comment (janethavi) [inline src/app/components/components.component.ts]: Shall we remove this extra line?

**Review by hasalfernando — COMMENTED** [review]
Nice work @rajitha1998 

I found some things that needs to be adjusted in the project cards.

### Cake-a-licious Website card
Change the name Codejunction to "codeJunction".

### Ajith motors Entry System card
Change "according to the type of account(staff/admin)" to "according to the type of account (staff/admin)".

### Fhix45 card
Remove the first sentence "This is one of the projects which was undertaken from codeJunction".

### CodeJunction Website card
Change the name "CodeJunction" to "codeJunction" in both the title and the description.


Shall we update these changes?

File: src/app/components/our-work/our-work.component.html [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+                                github: project.github|| null})">
+                            See more</button>
+                    </div>
+                </div>
+                <button type="button" class="slider_button btn btn-danger btn-just-icon" (click)="foward()"><i
+                        class="fa fa-angle-right"></i></button>
+            </div>
+        </div>
+    </div>
+</div>
```
Comment (janethavi) [inline src/app/components/our-work/our-work.component.html]: Shall we add the EOF line here?

File: src/assets/json/projects.json [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    "isWeb": "true"
+  },
+  {
+    "projectName": "Fire call",
+    "image": "firecall.jpg",
+    "description": "A sound wave classifying tool which analyse audio content. (Open Source ❤️)",
+    "largeDescription": "Fire call is an open-source project which is developed using python and flask framework. This project aims to classify sound waves through analysing audio content. We welcome everyone to contribute in this project. To get started please visit our Github repository.",
+    "github": "https://github.com/codeJunctionTeam/fire-call"
+  }
+]
```
Comment (janethavi) [inline src/assets/json/projects.json]: Shall we add the EOF line here?

**Review by janethavi — CHANGES_REQUESTED** [review]

**Comment by janethavi** [comment]
Great work @rajitha1998! Nicely done🚀

**Review by janethavi — APPROVED** [review]

**Review by hasalfernando — APPROVED** [review]
Well done @rajitha1998 Nice work!

**Pull request merged**